src/nonlinear.jl

```diff
@@ -600,10 +606,9 @@ function MathProgBase.optimize!(m::PajaritoModel)
             start_nlp = time()
             MathProgBase.optimize!(inf_model)
             cputime_nlp += time() - start_nlp
-            if MathProgBase.status(inf_model) == :Infeasible
-                (m.verbose > 2) && println("INF NLP Infeasible")
-                m.status = :Infeasible
-                return
+            inf_model_status = MathProgBase.status(inf_model)
+            if inf_model_status != :Optimal && inf_model_stauts != :Suboptimal
+                error("NLP Solver Failure.")
```

Comment:
We should print a warning an return ``:Error`` status, not crash the program with ``error()``